src/providers/settings.tsx

```diff
@@ -1,47 +1,79 @@
-import React, { createContext, FC, useContext } from 'react';
+import React, { createContext, FC, useCallback, useContext, useEffect, useMemo, useState } from 'react';
+import { LocalStorageService } from '../api/services/localStorageService';
 import { SpoilerLevel } from '../models/spoilerLevel';
 import { TagClassification } from '../models/tagClassification';
+import { KEY_VIEW_SETTINGS } from '../utils/localStorageKeys';
 
 interface Settings {
 
   /**
    * What tags categories to show.
    */
-  showTags: Record<TagClassification, boolean>;
+  readonly showTags: Readonly<Record<TagClassification, boolean>>;
 
   /**
    * Spoiler level for tags.
    */
-  spoilerLevel: SpoilerLevel;
+  readonly spoilerLevel: SpoilerLevel;
 
   /**
    * Whether nsfw content allowed or not.
    */
-  isNsfwContentAllowed: boolean;
+  readonly isNsfwContentAllowed: boolean;
+}
+
+interface SettingsSetters {
+
+  /** Updates settings. */
+  readonly updateSettings: (newSettings: Settings) => void;
```

Comment:
Actually, you could combine it with `Settings` interface, make `updateSettings` optional, so you wouldn't get errors on `createContext` function call with default values
After that you can assert that `updateSettings` is not `undefined` in `useSettingsContext` hook